packages/graphiql/src/components/GraphiQL.tsx

```diff
@@ -545,11 +547,15 @@ export function GraphiQLInterface(props: GraphiQLInterfaceProps) {
                             {tab.title}
                           </Tab.Button>
                           <Tab.Close
-                            onClick={() => {
-                              if (editorContext.activeTabIndex === index) {
-                                executionContext.stop();
+                            onClick={async () => {
+                              if (
```

Comment:
i think we should move all of this into `closeTab` itself, even what was directly in the handler here before, making it async. i see no reason to let this be an implementation detail for sdk users, as if the context they are using has this logic they would expect this as as an internal detail of the method if it's exposed to the user yeah?

```ts
  const closeTab = useCallback<EditorContextType['closeTab']>(
    async index => {
      if (await closeTabConfirmation(index)) {
      // this could be a breaking change for users who are already stopping 
      // execution like graphiql does here:
        if (index === tabState.activeTabIndex) {
          stop();
        }
        setTabState(current => {
          const updated = {
            tabs: current.tabs.filter((_tab, i) => index !== i),
            activeTabIndex: Math.max(current.activeTabIndex - 1, 0),
          };
          storeTabs(updated);
          setEditorValues(updated.tabs[updated.activeTabIndex]);
          onTabChange?.(updated);
          return updated;
        });
      }
    },
    [
      onTabChange,
      setEditorValues,
      storeTabs,
      closeTabConfirmation,
      tabState.activeTabIndex,
    ],
  );
  ```
and then here in graphiql:

```tsx

 <Tab.Close
    onClick={async () => this.closeTab(index)}
/>

```

Reply:
the reason why it was not inside `closeTab` in `ExecutionContextProvider` is that `ExecutionContextProvider` is children of `EditorContextProvider` and can be accessible from `EditorContextProvider`
<img width="433" alt="image" src="https://github.com/user-attachments/assets/37606086-8f5e-4715-85c1-dd5e36344af4">